src/teleop/gui/src/components/SAWaypointEditor.vue

```diff
@@ -0,0 +1,197 @@
+          E
+        </div>
+        <br>
+        <div style="display:inline-block">
+          <button v-on:click="addWaypoint(input)">Add Waypoint</button>
+        </div>
+      </div>
+      <div class="box1">
+        <div class="all-waypoints">
+          <h4 class="waypoint-headers">Waypoints</h4>
+          <button v-on:click="clearWaypoint">Clear Waypoints</button>
+        </div>
+        <draggable v-model="storedWaypoints" class="dragArea" draggable=".item'">
+          <WaypointItem v-for="waypoint, i in storedWaypoints" :key="i" v-bind:waypoint="waypoint" v-bind:index="i" v-on:delete="deleteItem($event)" v-on:find="findWaypoint($event)"/>
+        </draggable>
+      </div>
+    </div>
+  </template>
+  
+  <script>
+  import draggable from 'vuedraggable'
+  import {convertDMS} from '../utils.js';
+  import WaypointItem from './SAWaypointItem.vue'
+  import Checkbox from './Checkbox.vue'
+  import {mapMutations, mapGetters} from 'vuex'
+  import _ from 'lodash';
+  import L from 'leaflet'
+  export default {
+    data () {
+      return {
+        name: "Waypoint",
+        odom_format_in: 'DM',
+        input: {
+          lat: {
+            d: 0,
+            m: 0,
+            s: 0
+          },
+          lon: {
+            d: 0,
+            m: 0,
+            s: 0
+          }
+        },
+        storedWaypoints: []
+      }
+    },
+    methods: {
+      ...mapMutations('sa',{
+        setWaypointList: 'setWaypointList',
+        setHighlightedWaypoint: 'setHighlightedWaypoint',
+        setOdomFormat: 'setOdomFormat'
+      }),
+      deleteItem: function (payload) {
+        if(this.highlightedWaypoint == payload.index){
+          this.setHighlightedWaypoint(-1)
+        }
+        this.storedWaypoints.splice(payload.index, 1)
+      },
+      addWaypoint: function (coord) {
+        this.storedWaypoints.push({
+          name: this.name,
+          lat: (coord.lat.d + coord.lat.m/60 + coord.lat.s/3600),
+          lon: (coord.lon.d + coord.lon.m/60 + coord.lon.s/3600)
+        });
+      },
+      findWaypoint: function (payload) {
+        if(payload.index === this.highlightedWaypoint){
+          this.setHighlightedWaypoint(-1)
+        }
+        else{
+          this.setHighlightedWaypoint(payload.index)
+        }
+      },
+      clearWaypoint: function () {
+        this.storedWaypoints = [];
+      }
+    },
+    watch: {
+      storedWaypoints: function (newList) {
+        const waypoints = newList.map((waypoint) => {
+          return { latLng: L.latLng(waypoint.lat, waypoint.lon), name: waypoint.name };
+        });
+        this.setWaypointList(waypoints);
+      },
+      odom_format_in: function (newOdomFormat) {
+        this.setOdomFormat(newOdomFormat);
+        this.input.lat = convertDMS(this.input.lat, newOdomFormat);
+        this.input.lon = convertDMS(this.input.lon, newOdomFormat);
+        this.storedWaypoints.map((waypoint) => {
```

Comment:
I'm pretty sure this causes the waypoint editor to display NULL when you switch from D to DM and DM to DMS, etc. Was the waypoint editor tested?

Reply:
Erik and I actually switched it to the ERD one since they are functionally the same. But we definitely should clean up this set of files now. The waypoint editor did work though after switching to the ERD ones.

Reply:
Should we fix the SA waypoint files or just delete them and use the ERD ones?